.github/workflows/basic.yaml

```
@@ -0,0 +1,44 @@
+name: basic build and unit tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+
+  # shim basic test
+  shim:
+    name: shim basic test
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        go-version: [1.16.x, 1.17.x]
+
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - uses: actions/checkout@v2
+
+      - name: Run build
+        run: |
+          make binaries
+        working-directory: src/shim
+
+      - name: Run unit test
+        run: |
+          go test -v ./...
+        working-directory: src/shim
+
+      - name: Run go fmt check
+        uses: Jerome1337/gofmt-action@v1.0.4
+        with:
+          gofmt-path: 'src/shim'
+          gofmt-flags: '-l -d'
```

My preference would be `golangci-lint` with the appropriate config:
```suggestion
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3
        with:
          version: v1.48.0
          args: -v --timeout 5m
          working-directory: 'src/shim'
```